### PR TITLE
Complete Vet Clinic Database: Animal Table, Data and Queries

### PR DESCRIPTION
**Pull request by ometman**
**I have completed the Vet Clinic Database**
- The animal table, data insertion and queries have been implemented on branch 'animals-table'
- PostgreSql was downloaded and installed and run locally
- The Microverse templated was used to create the repo -https://github.com/microverseinc/curriculum-template-databases

**- The vet_clinic database and animals table was created with the following columns:**
   - id: integer, name: string, date_of_birth: date, escape_attempts: integer, neutered: boolean, weight_kg: decimal
- The table sql was copied into the schema.sql file.

**- The following data was inserted into the table and the sql was copied into the data.sql file.**
  -   Animal: His name is Agumon. He was born on Feb 3rd, 2020, and currently weighs 10.23kg. He was neutered and he has never tried to escape.
  -   Animal: Her name is Gabumon. She was born on Nov 15th, 2018, and currently weighs 8kg. She is neutered and she has tried to escape 2 times.
  -   Animal: His name is Pikachu. He was born on Jan 7th, 2021, and currently weighs 15.04kg. He was not neutered and he has tried to escape once.
  -   Animal: Her name is Devimon. She was born on May 12th, 2017, and currently weighs 11kg. She is neutered and she has tried to escape 5 times.

**- The following queries were written and the sql were copied into the queries.sql file:*
  1.   All animals whose name ends in "mon".
  2.   The name of all animals born between 2016 and 2019.
  3.   List of name of all animals that are neutered and have less than 3 escape attempts.
  4.   List of the date of birth of all animals named either "Agumon" or "Pikachu".
  5.   List of name and escape attempts of animals that weigh more than 10.5kg
  6.   All animals that are neutered.
  7.   All animals not named Gabumon.
  8.   All animals with weight between 10.4kg and 17.3kg (including animals with weights that equals precisely 10.4kg or 17.3kg)

**A screenshot of the results the queries are added as below .**
queries-1-4

queries-4-8

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -1,3 +1,10 @@
...
 
-SELECT * from animals WHERE name = 'Luna';
+SELECT * from animals WHERE name LIKE '%mon';
+SELECT name from animals WHERE date_of_birth BETWEEN '2016-01-01' AND '2019-12-31';
+SELECT name from animals WHERE neutered = TRUE AND escape_attempts < 3;
+SELECT date_of_birth FROM animals WHERE name = 'Agumon' or name = 'Pikachu';
+SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.50;
+SELECT * FROM animals WHERE neutered = TRUE;
+SELECT * from animals WHERE name <> 'Gabumon';
+SELECT * FROM animals WHERE weight_kg >= 10.40 OR weight_kg <= 17.30;
```
Comment (V-Blaze) [inline queries.sql]: - You have done great so far :+1: :100: But there is an issue you need to fix before moving to the next milestone. :+1: 

- - Screenshot reference:
Selection_026

- [ ] Your current implementation of the last query is not correct. The question says `animals with a weight between 10.4kg and 17.3kg `, In your current query, you are selecting all rows from the "animals" table where the "weight_kg" is greater than or equal to 10.40 OR the "weight_kg" is less than or equal to 17.30. This query will return all rows that meet either of these conditions, which means it includes rows with "weight_kg" values both below 10.40 and above 17.30.
- In summary, Your current query uses the OR operator, which results in a broader range of rows that meet either condition, while the correct query should use the AND operator, which narrows down the result set to rows that meet both conditions simultaneously, falling within the specified weight range. :+1: :100: :+1: I'll suggest you update this and also update your screenshot results. :smile: :small_airplane: 

the correct implementation should use the `AND` operator:
```ruby
SELECT * FROM animals WHERE weight_kg >= 10.40 AND weight_kg <= 17.30; 
```

**Review by V-Blaze — CHANGES_REQUESTED** [review]
## Hi @ometman 

## Required changes 🖊️ 
# Almost there 🟡 🟦 🟨 

giphy


Great work so far, You have come a long way, keep moving forward👏 👍  
You've done well implementing the project requirements, but there are still some issues I think you need to improve on. 😄 👋 


## Highlights
- Table created correctly and correct data inseted💯
-  Querry result screenshots added to pull requests👍
- Most queries written correctly 🟢 👍

 ## Required Changes ♻️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (ometman) [reply queries.sql]: Sure, you are right. Thanks 👍

**Review by Alfredbis29 — APPROVED** [review]
# STATUS: APPROVED   
 ## Hi  @ometman 
Your project is complete! There is nothing else to say other than... Great job clap Congratulations! :tada:
 
 TO HIGHLIGHT:
 
 * Your project is professional :white_check_mark:
 * Good Job :+1:
 
 Now you can merge this branch and move on 1st_place_medal: HAPPY CODING AND GOOD LUCK WITH THE COMING ONES :crossed_fingers: :


 
 ## Optional suggestions

- [x] N/A

 _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._
 
Cheers and Happy coding! :clap:  :clap: :clap:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me @Alfredbis29 in your question so I can receive the notification.
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
